src/psyclone/psyGen.py

```diff
@@ -6093,50 +6198,71 @@ def __init__(self, kernel=None):
-                           reference to an integer symbol with the extent. \
-                           If it is an empty list then the symbol represents \
-                           a scalar.
-        :param str scope: It is 'local' if the symbol just exists inside the \
-                          kernel scope or 'global_*' if the data survives \
-                          outside of the kernel scope. Note that \
-                          global-scoped symbols also have postfixed \
-                          information about the sharing mechanism, at the \
-                          moment just 'global_argument' is available for \
-                          variables passed in/out of the kernel by argument.
-        :param bool is_input: Whether the symbol represents data that exists \
-                              before the kernel is entered and that is passed \
-                              into the kernel.
-        :param bool is_output: Whether the symbol represents data that is \
-                               survives outside the kernel upon exit.
-        :raises KeyError: The provided name can not be used as key in the
-                          table.
-        '''
-        if name in self._symbols:
-            raise KeyError("Symbol table already contains a symbol with"
-                           " name '{0}'.".format(name))
+    def declare(self, new_symbol):
+        '''Declare a new symbol in the symbol table.
+
+        :param new_symbol: The symbol to add to the symbol table.
+        :type new_symbol: :py:class:`psyclone.psyGen.Symbol`
 
-        self._symbols[name] = Symbol(name, datatype, shape, scope, is_input,
-                                     is_output)
+        :raises KeyError: If the symbol name is already in use.
+
+        '''
+        if new_symbol.name in self._symbols:
+            raise KeyError("Symbol table already contains a symbol with"
+                           " name '{0}'.".format(new_symbol.name))
+        self._symbols[new_symbol.name] = new_symbol
+
+    def swap_symbol_properties(self, symbol1, symbol2):
+        '''Swaps the properties of symbol1 and symbol2 apart from the symbol
+        name. Argument list positions are also updated appropriately.
+
+        :param symbol1: The first symbol.
+        :type symbol1: :py:class:`psyclone.psyGen.Symbol`
+        :param symbol2: The second symbol.
+        :type symbol2: :py:class:`psyclone.psyGen.Symbol`
+
+        :raises KeyError: If either of the supplied symbols are not in \
+        the symbol table.
+        :raises TypeError: if the supplied arguments are not symbols, \
+        or the names of the symbols are the same in the SymbolTable \
+        instance.
+
+        '''
+        for symbol in [symbol1, symbol2]:
+            if not isinstance(symbol, Symbol):
+                raise TypeError("Arguments should be of type 'Symbol' but "
+                                "found '{0}'.".format(type(symbol).__name__))
```

Comment:
There are two reasons for doing this.

1) It makes for a nicer error. If we just print the type of something then we get <type 'classname'>. If we add __name__ we just get the class name.

2) If you print the type of something in Python2 and in Python3 you can get different output - it can be called type in one case and class in another. By using __name__ we get the same output in either Python2 or 3, which is nice both for consistency and for unit tests.

Reply:
TIL. I will use it from now on.